### PR TITLE
feat(devices): Add metrics on device updates, and a flag to disable them

### PR DESCRIPTION
**Pull request by rfk**
To help us with rolling out device metadata updates to release Firefox, this PR adds two things:

* More metrics on what has changed (if anything) to trigger the update
* A safety switch to let us easily pref the feature off if we get swamped

@jrgm what do you think?

### CODE REVIEW

**Comment by rfk** [comment]
@pb could you please r? for the extra logging etc here?

File: lib/routes/account.js [hunk 1]
```diff
@@ -845,15 +845,39 @@ module.exports = function (
...
+            log.increment('device.update.type')
+          }
+          if (payload.pushCallback && payload.pushCallback !== sessionToken.deviceCallbackURL) {
+            spurious = false
+            log.increment('device.update.pushCallback')
+          }
+          if (payload.pushPublicKey && payload.pushPublicKey !== sessionToken.deviceCallbackPublicKey) {
+            spurious = false
+            log.increment('device.update.pushPublicKey')
+          }
```
Comment (philbooth) [inline lib/routes/account.js]: Purely for readability, is it worth considering extracting the above to, say, an `isSpuriousUpdate` function?

File: test/local/account_routes.js [hunk 2]
```diff
@@ -213,3 +213,146 @@ test(
...
+      getRoute(accountRoutes, '/account/device')
+        .handler(mockRequest, function(response) {
+          resolve(response)
+        })
+    })
+    .then(function(response) {
+      t.equal(mockDB.updateDevice.callCount, 0, 'updateDevice was not called')
+
+      t.equal(mockLog.increment.callCount, 1, 'a counter was incremented')
+      t.equal(mockLog.increment.firstCall.args[0], 'device.update.spurious')
```
Comment (philbooth) [inline test/local/account_routes.js]: Wow, I never noticed these local route tests before, are they new? I should have used them for the KPI activty event stuff, will make sure I do so from now on.

**Comment by philbooth** [comment]
Looks good, **r+**.

File: test/local/account_routes.js (hunk 2 above)
Comment (rfk) [inline test/local/account_routes.js]: They're newish, and growing quickly, which I think is a good sign that they're valuable :-)

**Comment by rfk** [comment]
updated with the suggested refactor, will merge if travis gives the ok

**Pull request merged**